h2/src/main/org/h2/command/dml/Select.java

```diff
@@ -193,12 +193,12 @@ public Object getCurrentGroupExprData(Expression expr) {
      */
     public void setCurrentGroupExprData(Expression expr, Object obj) {
         if (cachedLookup == expr) {
-            assert currentGroupByExprData[cachedLookupIndex] != null;
+            assert currentGroupByExprData[cachedLookupIndex] == null;
             currentGroupByExprData[cachedLookupIndex] = obj;
         }
         Integer index = exprToIndexInGroupByData.get(expr);
         if (index != null) {
-            assert currentGroupByExprData[index] != null;
+            assert currentGroupByExprData[index] == null;
```

Comment:
Not here, this assertion was good.

Reply:
I dont see how - if we setting a new value, there should be null in that entry, since nowhere do we ever overwrite a previous value

Reply:
If it was not set we can't enter this condition, probably it can be removed, but there is another big problem with empty map and non-null cached expression. I already mentioned it in another comment.

Reply:
This method can be entered with the same expressions in cache and in parameter, but with empty map of all expressions.

Cached expression should be null if map of expressions is empty, but it's not sometimes.